### ISSUE TITLE
manifest for prefecthq/prefect:v2.14.6-python3.11 not found: manifest unknown:

### ISSUE THREAD

**Issue by theompm** [PrefectHQ/prefect#11259]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

Hey prefect team.

i found a bug in` prefect --version`: 

2.14.0 -> 2.14.0
2.14.6 -> **v**2.16.6

this leads to issue when `prefect deploy`:

```
prefect.utilities.dockerutils.BuildError: manifest for prefecthq/prefect:v2.14.6-python3.11 not found: manifest unknown: manifest unknown
```

# Prefect Version 2.14.0

```
prefect --version 
2.14.0
```

## When

```yaml
# Generic metadata about this project
name: example_project
prefect-version: 2.14.0

# build section allows you to manage and build docker images
build:
- prefect_docker.deployments.steps.build_docker_image:
    id: build_image
    requires: prefect-docker>=0.3.1
    image_name: example/image
    tag: 0.1.1
    dockerfile: auto

```

```
prefect deploy

build completes
```

# Prefect Version 2.14.6

```
prefect --version 
v2.14.6
```

## When

```yaml
# Generic metadata about this project
name: example_project
prefect-version: 2.14.6

# build section allows you to manage and build docker images
build:
- prefect_docker.deployments.steps.build_docker_image:
    id: build_image
    requires: prefect-docker>=0.3.1
    image_name: example/image
    tag: 0.1.1
    dockerfile: auto

```

```
prefect deploy
```
ERROR:
```error
prefect.utilities.dockerutils.BuildError: manifest for prefecthq/prefect:v2.14.6-python3.11 not found: manifest unknown: manifest unknown
```

### Reproduction

```python3
# Prefect Version 2.14.0


prefect --version 
2.14.0
```

## When

# Generic metadata about this project
name: example_project
prefect-version: 2.14.6

# build section allows you to manage and build docker images
build:
- prefect_docker.deployments.steps.build_docker_image:
    id: build_image
    requires: prefect-docker>=0.3.1
    image_name: example/image
    tag: 0.1.1
    dockerfile: auto

```

```
prefect deploy

build completes
```


# Prefect Version 2.14.6

```
prefect --version 
v2.14.6
```

## When

```yaml
# Generic metadata about this project
name: example_project
prefect-version: 2.14.6

# build section allows you to manage and build docker images
build:
- prefect_docker.deployments.steps.build_docker_image:
    id: build_image
    requires: prefect-docker>=0.3.1
    image_name: example/image
    tag: 0.1.1
    dockerfile: auto

```

```
prefect deploy
```

```error
prefect.utilities.dockerutils.BuildError: manifest for prefecthq/prefect:v2.14.6-python3.11 not found: manifest unknown: manifest unknown
```
```


### Error

```python3
Step 1/6 : FROM prefecthq/prefect:v2.14.6-python3.11
Traceback (most recent call last):
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/deployments/steps/core.py", line 156, in run_steps
    step_output = await run_step(step, upstream_outputs)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/deployments/steps/core.py", line 127, in run_step
    result = await from_async.call_soon_in_new_thread(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/_internal/concurrency/calls.py", line 293, in aresult
    return await asyncio.wrap_future(self.future)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/_internal/concurrency/calls.py", line 318, in _run_sync
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/_internal/compatibility/deprecated.py", line 155, in wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect_docker/deployments/steps.py", line 194, in build_docker_image
    raise BuildError(event["error"])
prefect.utilities.dockerutils.BuildError: manifest for prefecthq/prefect:v2.14.6-python3.11 not found: manifest unknown: manifest unknown

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/cli/_utilities.py", line 41, in wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/utilities/asyncutils.py", line 255, in coroutine_wrapper
    return call()
           ^^^^^^
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/_internal/concurrency/calls.py", line 398, in __call__
    return self.result()
           ^^^^^^^^^^^^^
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/_internal/concurrency/calls.py", line 284, in result
    return self.future.result(timeout=timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/_internal/concurrency/calls.py", line 168, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/saficu/.conda/envs/python3.11/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/_internal/concurrency/calls.py", line 355, in _run_async
    result = await coro
             ^^^^^^^^^^
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/cli/deploy.py", line 297, in deploy
    await _run_single_deploy(
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/client/utilities.py", line 51, in with_injected_client
    return await fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/cli/deploy.py", line 601, in _run_single_deploy
    await run_steps(build_steps, step_outputs, print_function=app.console.print)
  File "/home/saficu/theo/projects/datalake_to_his/.venv/lib/python3.11/site-packages/prefect/deployments/steps/core.py", line 184, in run_steps
    raise StepExecutionError(f"Encountered error while running {fqn}") from exc
prefect.deployments.steps.core.StepExecutionError: Encountered error while running prefect_docker.deployments.steps.build_docker_image
```

### Versions

```Text
Version:             v2.14.6
API version:         0.8.4
Python version:      3.11.5
Git commit:          b99932a4
Built:               Wed, Nov 22, 2023 3:19 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         cloud
```

**Comment by zhen0** [comment]
Thanks for the issue @theompm - we believe this came from an incorrect tag in the 2.14.6 release.  We're planning a release of 2.14.7 today to minimize the impact of this.

**Comment by desertaxle** [comment]
This has been fixed in the 2.14.8 release. If you run into this issue, please upgrade to 2.14.8!
